dicompylercore/dose.py

```diff
@@ -0,0 +1,283 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# dose.py
+"""Class for summing DICOM-RT Dose grid data."""
+# Copyright (c) 2009-2016 Aditya Panchal
+# Copyright (c) 2019-2020 Dan Cutright
+# This file is part of dicompyler-core, released under a BSD license.
+#    See the file license.txt included with this distribution, also
+#    available at https://github.com/dicompyler/dicompyler-core/
+#
+# This code was adapted from dicom_dose_sum.py from DVH Analytics:
+#    https://github.com/cutright/DVH-Analytics/blob/master/dvha/tools/dicom_dose_sum.py
+
+from copy import deepcopy
+import numpy as np
+from scipy.ndimage import map_coordinates
```

Comment:
Should add the `scipy_available` check either here or in the interpolation functions themselves. We may only want it to show up if they aren't doing a direct sum.